### PR TITLE
DAOS-5832 test: Update get_partition_hosts

### PR DESCRIPTION
**Pull request by mjean308**
Include the slurm reservation as an input to get_partition_hosts()
If a slurm reservation exists verify the nodes are in the specified partition and
return a list of nodes in the reservation.

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 1]
```diff
@@ -485,7 +497,7 @@ def get_config_file(self, name, command):
         filename = "{}_{}_{}.yaml".format(self.config_file_base, name, command)
         return os.path.join(self.tmp, filename)
 
-    def add_agent_manager(self, config_file=None, common_cfg=None, timeout=15):
+    def add_agent_manager(self, config_file=None, common_cfg=None, timeout=5):
```
Comment (mjean308) [inline src/tests/ftest/util/apricot/apricot/test.py:500]: will fix this...  timeout should be 15

**Review by shimizukko — APPROVED** [review]
LGTM

**Comment by mjean308** [comment]
Single test failure is due to DAOS-5835

**Review by phender — APPROVED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM
If another update is needed "client_reservation:" in yaml files can be commented as it will be None anyways.

**Pull request merged**